server/autolinkplugin/plugin.go

```diff
@@ -136,86 +138,83 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
-			startPos, endPos = textNode.Range.Position+offset, textNode.Range.End+offset
-			origText = postText[startPos:endPos]
-			if textNode.Text != origText {
-				p.API.LogError(fmt.Sprintf("Markdown text did not match range text, '%s' != '%s'", textNode.Text,
-					origText))
+
+		case *markdown.Text:
+			start, end = node.Range.Position+offset, node.Range.End+offset
+			toProcess = message[start:end]
+			if node.Text != toProcess {
+				p.API.LogDebug("skipping text: parsed markdown did not match original", "parsed", node.Text, "original", toProcess, "post_id", post.Id)
 				return true
 			}
 		}
 
-		if origText != "" {
-			newText := origText
-
-			for _, link := range conf.Links {
-				if !p.inScope(link.Scope, channelName, teamName) {
-					continue
-				}
+		if toProcess == "" {
+			return true
+		}
 
-				out := link.Replace(newText)
-				if out == newText {
-					continue
-				}
+		processed := toProcess
+		for _, link := range conf.Links {
+			if !p.inScope(link.Scope, channelName, teamName) {
+				continue
+			}
 
-				if !link.ProcessBotPosts {
-					if author == nil && authorErr == nil {
-						author, authorErr = p.API.GetUser(post.UserId)
-						if authorErr != nil {
-							// NOTE: Not sure how we want to handle errors here, we can either:
-							// * assume that occasional rewrites of Bot messges are ok
-							// * assume that occasional not rewriting of all messages is ok
-							// Let's assume for now that former is a lesser evil and carry on.
-							p.API.LogError("failed to check if message for rewriting was send by a bot", "error", authorErr)
-						}
-					}
+			out := link.Replace(processed)
+			if out == processed {
+				continue
+			}
 
-					if author != nil && author.IsBot {
-						continue
+			if !link.ProcessBotPosts {
+				if author == nil && authorErr == nil {
+					author, authorErr = p.API.GetUser(post.UserId)
+					if authorErr != nil {
+						// NOTE: Not sure how we want to handle errors here, we can either:
+						// * assume that occasional rewrites of Bot messges are ok
+						// * assume that occasional not rewriting of all messages is ok
+						// Let's assume for now that former is a lesser evil and carry on.
+						p.API.LogError("failed to check if message for rewriting was send by a bot", "error", authorErr)
 					}
 				}
 
-				newText = out
-			}
-			if origText != newText {
-				postText = postText[:startPos] + newText + postText[endPos:]
-				offset += len(newText) - len(origText)
+				if author != nil && author.IsBot {
+					continue
+				}
 			}
+
+			processed = out
+		}
+
+		if toProcess != processed {
+			message = message[:start] + processed + message[end:]
+			offset += len(processed) - len(toProcess)
+			changed = true
 		}
 
 		return true
 	})
-	if post.Message != postText {
-		post.Message = postText
-	}
-
-	post.Hashtags, _ = model.ParseHashtags(post.Message)
 
+	if changed {
+		post.Message = message
+		post.Hashtags, _ = model.ParseHashtags(message)
```

Comment:
This is now inside the if, looks like something that makes sense, because if I haven't change it, the original post should be having properly processes the hashtags already, but is a subtle change in the behavior and I just want to let you know to ensure that it is on purpose change.

Reply:
Exactly. I think it was basically a bug/redundant before.